bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/serialization/InstantDeserializer.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hue.internal.serialization;
+
+import java.lang.reflect.Type;
+import java.time.Instant;
+import java.time.format.DateTimeParseException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+
+/**
+ * The {@link InstantDeserializer} converts a formatted UTC string to {@link Instant}.
+ *
+ * @author Jacob Laursen - Initial contribution
+ */
+@NonNullByDefault
+public class InstantDeserializer implements JsonDeserializer<Instant> {
```

Comment:
Minor point, but for better clarity perhaps this class should be located in the `internal/dto/clip2/helper` module. Or??

Reply:
Well, it's neither a DTO, CLIP 2-specific or a helper. However, perhaps we could create a subdirectory, _internal/api_ and move the _dto_ directory here as well as _serialization_. WDYT?

Additionally, it would be nice to be able to collapse CLIP 1 DTO classes.

So the proposed structure would be something like this:
- internal/api
  - dto
    - clip1
    - clip2
  - serialization

To limit scope and reduce conflicts, I would propose to do this reorganization in a follow-up PR with only this refactoring.